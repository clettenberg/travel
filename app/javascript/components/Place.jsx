import React from 'react';
import PropTypes from 'prop-types';

const Place = props => (
  <div className="card">
    <div className="card-body d-flex justify-content-between">
      <div id="name">{props.name}</div>
    </div>
  </div>
);

Place.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Place;
