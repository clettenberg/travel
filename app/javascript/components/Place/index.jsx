import React from 'react'
import Map from '../Map'

import styles from './style.module.scss'

const Place = ({ name, date, note, point }) => (
  <React.Fragment>
    <div className={styles.container}>
      {point &&
        <div style={{ flexGrow: 2 }}>
          <Map points={[point]} center={point} zoom={15} />
        </div>
      }
    </div>
    <div className={styles.content}>
      <h1 className='display-4 text-truncate'>
        {name}
      </h1>
      <h6>
        {date}

        <a
          href={window.location.pathname.concat('/edit')}
        >
          Edit
        </a>
      </h6>

      <hr />

      <div className='row'>
        <div className={`${styles.note} trix-content col-md-6 col-sm-12`}>
          {note || <p className='text-secondary font-weight-light font-italic'>
            No note yet.
          </p>
          }
        </div>
      </div>
    </div>
  </React.Fragment>
)

export default Place
