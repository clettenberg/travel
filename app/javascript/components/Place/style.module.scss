$mobile-breakpoint: 768px;

.container {
  div[class*=this-is-the-map] {
    height: 500px;
    width: auto;
  }
}
