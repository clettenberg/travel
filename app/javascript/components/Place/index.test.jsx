
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Place from './index'

describe('Place', () => {
  it('should render', () => {
    const wrapper = shallow(<Place />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
