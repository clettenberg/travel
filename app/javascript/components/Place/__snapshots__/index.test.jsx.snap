// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Place should render 1`] = `
<Fragment>
  <div />
  <div>
    <h1
      className="display-4 text-truncate"
    />
    <h6>
      <a
        className="btn btn-primary btn-sm"
        href="//edit"
      >
        Edit
      </a>
    </h6>
    <hr />
    <div
      className="row"
    >
      <div
        className="undefined trix-content col-md-6 col-sm-12"
      >
        <p
          className="text-secondary font-weight-light font-italic"
        >
          No note yet.
        </p>
      </div>
    </div>
  </div>
</Fragment>
`;
